### PR TITLE
Add pagination to weapon swap UI

### PR DESCRIPTION
**Pull request by arquelion**
## Describe your changes

- Add pagination to the weapon swap UI
- Add a required `namespace` prop to `ShowingTextProps` and modified `ShowingItem` to additionally pass `ns={namespace}` to the `Trans` component and updated all affected locations - This ensures the translation is fetched from the correct namespace as a couple pages have a translation unit containing several namespaces which contain the same entry.
- Minor aesthetic changes to the bottom of the artifact swap UI

## Issue or discord link

- 

## Testing/validation

image
image
I've also manually inspected the main artifact/weapon/character pages as well as the artifact swap UI to confirm the "Showing..." text is correct.

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [x] I have commented my code in hard-to understand areas.
- [x] I have made corresponding changes to README or wiki.
- [x] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Comment by frzyc** [comment]
@arquelion there was a conflict on this branch due to a recent change. I've resolved it for you.

**Review by frzyc — APPROVED** [review]

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/WeaponSwapModal.tsx [hunk 1]
```diff
@@ -116,6 +132,42 @@ export default function WeaponSwapModal({
     [dbDirty, database, rarity, weaponTypeKey, deferredSearchTerm]
   )
 
+  const { weaponIdsToShow, numPages, currentPageIndex } = useMemo(() => {
```
Comment (nguyentvan7) [inline apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/WeaponSwapModal.tsx:135]: Looks like we could make this a util to be shared between arti/weapon/char pagination?

Comment (nguyentvan7) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/WeaponSwapModal.tsx:135]: I'm also ok just saving that for a different PR

**Review by nguyentvan7 — APPROVED** [review]
Didn't do functional testing

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/WeaponSwapModal.tsx (hunk 1 above)
Comment (frzyc) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/WeaponSwapModal.tsx:135]: Lets do another PR later, I am moving some code around this area and would be nice to have less noise.

**Pull request merged**